libkbfs/tlf_handle_resolve.go

```diff
@@ -283,13 +303,15 @@ func (rsa resolvableSocialAssertion) resolve(ctx context.Context) (
 }
 
 // MakeTlfHandle creates a TlfHandle from the given tlf.Handle and the
-// given normalizedUsernameGetter (which is usually a KBPKI).
+// given normalizedUsernameGetter (which is usually a KBPKI). Note
+// that `t` could be different from `bareHandle.Type()`, if this is an
+// implicit team TLF.
```

Comment:
Could you explain what `t` is here? It seems to me it should have been `tlf.KeyingType` but I think I understood it wrong.

Reply:
It simply replaces `bareHandle.Type()` in the `makeTlfHandleHelper()` call below.  That type is unreliable because it would report any team-backed TLF as `tlf.SingleTeam`, since `tlf.Handle` doesn't have it's own type field.

It's used to set `tlfType` of the new `libkbfs.TlfHandle`, so it's definitely not supposed to be the keying type.

I'll update this comment to explain that a bit.

Reply:
Thanks for the explanation. I'm still a bit confused though. Why does `bareHandle.Type()` report iTeam TLFs as `tlf.SingleTeam`? Shouldn't that be what `TypeForKeying()` does?

Reply:
A `tlf.Handle` does not have a separate type field, and so it infers the type by whether or not the first writer is a team.  It's basically the same thing that `tlf.Handle.TypeForKeying()` does (except with the Keying types).  But for `Type()`, this isn't necessarily correct, because an i-team TLF which should either have a type of Private or Public would have a team as the first writer, and so would return `tlf.SingleTeam` as the type incorrectly.

Reply:
Ah that's right. Thanks!